MQ2GMCheck.cpp

```diff
@@ -100,129 +123,146 @@ class MQ2GMCheckType : public MQ2Type
 	virtual bool GetMember(MQVarPtr VarPtr, const char* Member, char* Index, MQTypeVar& Dest) override
 	{
 		using namespace mq::datatypes;
-
-		char szTmp[MAX_STRING] = { 0 };
-		unsigned int x;
-		long lResult;
 		MQTypeMember* pMember = MQ2GMCheckType::FindMember(Member);
+
 		if (!pMember)
 			return false;
+
 		switch ((GMCheckMembers)pMember->ID)
 		{
 		case GMCheckMembers::Status:
 			Dest.DWord = bGMCheck;
 			Dest.Type = pBoolType;
 			return true;
+
 		case GMCheckMembers::GM:
 			Dest.DWord = GMCheck();
 			Dest.Type = pBoolType;
 			return true;
+
 		case GMCheckMembers::Names:
-			szTmp[0] = 0;
-			for (x = 0; x < GMNames.size(); x++)
-			{
-				std::string& VectorRef = GMNames[x];
+		{
+			char szTmp[MAX_STRING] = { 0 };
+			for (std::string GMName : GMNames) {
```

Comment:
```suggestion
			for (const std::string& GMName : GMNames) {
```
to avoid copying strings every iteration.